compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -254,8 +254,5 @@ pub fn handle_native_features(sess: &Session) -> Vec<String> {
 }
 
 pub fn tune_cpu(sess: &Session) -> Option<&str> {
-    match sess.opts.debugging_opts.tune_cpu {
-        Some(ref s) => Some(handle_native(&**s)),
-        None => None,
-    }
+    sess.opts.debugging_opts.tune_cpu.as_ref().map(|s| handle_native(s))
```

Comment:
same here, I would use `Some(handle_native(sess.opts.debugging_opts.tune_cpu.as_ref()?))`

Reply:
That line is relatively long and you have to parse it in full before you can understand it, unlike the `map` version where you can basically follow along. Although that could be solved by introducing a variable like 
```
let s = sess.opts.debugging_opts.tune_cpu.as_ref()?;
Some(handle_native(s))
```

Reply:
Yea, an intermediate variable makes it more readable :+1:

Reply:
Although... does `sess.opts.debugging_opts.tune_cpu.as_ref().map(handle_native)` work?

Reply:
```
error[E0631]: type mismatch in function arguments
   --> compiler/rustc_codegen_llvm/src/llvm_util.rs:257:52
    |
204 | fn handle_native(name: &str) -> &str {
    | ------------------------------------ found signature of `for<'r> fn(&'r str) -> _`
...
257 |     sess.opts.debugging_opts.tune_cpu.as_ref().map(handle_native)
    |                                                    ^^^^^^^^^^^^^ expected signature of `fn(&std::string::String) -> _`
```

Reply:
ah yea, that explains the original `&**`, thanks for testing